### PR TITLE
Added New Generic new version and removed old versions

### PR DESCRIPTION
**Pull request by gammonpeter**

### CODE REVIEW

**Review by pjmonks — CHANGES_REQUESTED** [review]
I like how this component view has been generalised, it makes me wonder if something similar could be done to the model views.

My review comments are generally just tidying up if you could look at those please. Otherwise well done.

File: src/app/app-routing.module.ts [hunk 1]
```diff
@@ -135,14 +134,9 @@ export const pageRoutes: { states: Ng2StateDeclaration[] } = {
...
-      name: 'appContainer.mainApp.twoSidePanel.catalogue.newVersionDataModel',
-      url: '/newVersion/dataModel/:dataModelId',
-      component: NewVersionDataModelComponent
-    },
-    {
-      name: 'appContainer.mainApp.twoSidePanel.catalogue.newVersionReferenceDataModel',
-      url: '/newVersion/referenceDataModel/:referenceDataModelId',
-      component: NewVersionReferenceDataModelComponent
+      name: 'appContainer.mainApp.twoSidePanel.catalogue.newVersionModel',
+      url: '/newVersion/:id/:domainType',
```
[the remark this answers is not in the document]
Comment (gammonpeter) [reply src/app/app-routing.module.ts:138]: Updated to be consistent

File: src/app/app-routing.module.ts [hunk 2]
```diff
@@ -135,14 +134,9 @@ export const pageRoutes: { states: Ng2StateDeclaration[] } = {
...
     {
-      name: 'appContainer.mainApp.twoSidePanel.catalogue.newVersionDataModel',
-      url: '/newVersion/dataModel/:dataModelId',
-      component: NewVersionDataModelComponent
-    },
-    {
-      name: 'appContainer.mainApp.twoSidePanel.catalogue.newVersionReferenceDataModel',
-      url: '/newVersion/referenceDataModel/:referenceDataModelId',
-      component: NewVersionReferenceDataModelComponent
+      name: 'appContainer.mainApp.twoSidePanel.catalogue.newVersionModel',
```
[the remark this answers is not in the document]
Comment (gammonpeter) [reply src/app/app-routing.module.ts:137]: Updated and replaced with new generic

File: src/app/shared/new-version/new-version.component.ts [hunk 3]
```diff
@@ -16,27 +16,23 @@ limitations under the License.
...
+  selector: 'mdm-new-version',
+  templateUrl: './new-version.component.html',
+  styleUrls: ['./new-version.component.scss']
 })
-export class NewVersionReferenceDataModelComponent implements OnInit {
+export class NewVersionComponent implements OnInit {
+
   step = 1;
-  referenceDataModel: ReferenceDataModelDetail;
+  catalogueItem: any;
```
[the remark this answers is not in the document]
Comment (gammonpeter) [reply src/app/shared/new-version/new-version.component.ts:34]: Updated

File: package.json [hunk 4]
```diff
@@ -35,7 +35,7 @@
     "@bpmn-io/dmn-migrate": "^0.4.3",
     "@ctrl/ngx-codemirror": "4.0.1",
     "@fortawesome/fontawesome-free": "5.15.1",
-    "@maurodatamapper/mdm-resources": "git+https://github.com/MauroDataMapper/mdm-resources.git#develop",
+    "@maurodatamapper/mdm-resources": "^4.6.0-SNAPSHOT",
```
Comment (pjmonks) [inline package.json:38]: Why has this changed? I thought @olliefreeman said the npm reference to `mdm-resources` should remain pointing at the `develop` branch?

**Review by pjmonks — COMMENTED** [review]
Your feedback implementation is good. Can we just check what the `package.json` ref to `mdm-resources` should be first before I accept/merge?

File: package.json (hunk 4 above)
Comment (gammonpeter) [reply package.json]: yeah agreed i ran a npm updated so not sure why it changed

**Review by pjmonks — APPROVED** [review]

**Pull request merged**